plutus-ledger-api/src/PlutusLedgerApi/Common/Versions.hs

```diff
@@ -79,49 +95,72 @@ builtinsIntroducedIn = Map.fromList [
           ChooseData, ConstrData, MapData, ListData, IData, BData, UnConstrData, UnMapData, UnListData, UnIData, UnBData, EqualsData,
           MkPairData, MkNilData, MkNilPairData
           ]),
-  -- Vasil is protocolversion=7.0
   ((PlutusV2, vasilPV), Set.fromList [
           SerialiseData
           ]),
-  -- Chang is protocolversion=8.0
-  ((PlutusV2, changPV), Set.fromList [
+  ((PlutusV2, valentinePV), Set.fromList [
           VerifyEcdsaSecp256k1Signature, VerifySchnorrSecp256k1Signature
           ])
   ]
 
-{-| Query the protocol version that a specific ledger Plutus version was first introduced in.
+{-| A map indicating which Plutus Core versions were introduced in which
+'ProtocolVersion' and 'PlutusLedgerLanguage'. Each version should appear at most once.
```

Comment:
Then perhaps the version should be the map key?

Reply:
We could do something better here for this map indeed.

I think a first step  is to define its keys in terms of  `languageIntroducedIn` to avoid mistakes on PVs?

Reply:
Another idea is to turn this into a 3-assoc. list with type [(PlutusLedgerLanguage,ProtocolVersion,Version)] and either on compile time or runtime have a test that the x,y,z parts are unique:

```
plcVersionsIntroducedIn = [ (PlutusV1, alonzoPv, plcVersion100)  , ...]
(x,y,z) = unzip3 plcVersionsIntroducedIn
test that isUnique x , y , z 
isUnique l =  length l == nubOrd l (or some other more efficient impl.)
```

Reply:
We could also delve into some custom `IsList`  uniqueness checker that blows up with a runtime test? I am not sure how that would look like

Reply:
I mean we could simply reverse the key and the value for this map, and lookup by plutus core version.

Reply:
It's the same thing as for `builtinsIntroducedIn`. I'd rather keep them consistent and fix them together, if we can!